src/services/GLSL/vertex_shader_catalog.glsl

```diff
@@ -0,0 +1,212 @@
+#version 300 es
+precision highp float;
+
+#define BOX_FILLED 0
+#define BOX_LINED 1
+#define CIRCLE_FILLED 2
+#define CIRCLE_LINED 3
+#define HEXAGON_FILLED 4
+#define HEXAGON_LINED 5
+#define RHOMB_FILLED 6
+#define RHOMB_LINED 7
+#define TRIANGLE_FILLED_UP 8
+#define TRIANGLE_LINED_UP 9
+#define ELLIPSE_FILLED 10
+#define ELLIPSE_LINED 11
+#define TRIANGLE_FILLED_DOWN 12
+#define TRIANGLE_LINED_DOWN 13
+#define HEXAGON_FILLED_2 14
+#define HEXAGON_LINED_2 15
+#define CROSS_FILLED 16
+#define CROSS_LINED 17
+#define X_FILLED 18
+#define X_LINED 19
+#define LineSegment_FILLED 20
+
+#define PI radians(180.0)
+#define SQRT3 sqrt(3.0)
+#define SIN_60 0.86602540378
+
+#define LINEAR 0
+#define LOG 1
+#define SQRT 2
+#define SQUARE 3
+#define POWER 4
+#define GAMMA 5
+
+uniform sampler2D uCmapTexture;
+uniform sampler2D uXTexture;
+uniform sampler2D uYTexture;
+uniform sampler2D uSizeTexture;
+uniform sampler2D uColorTexture;
+uniform sampler2D uOrientationTexture;
+uniform highp usampler2D uSelectedSourceTexture;
+uniform sampler2D uSizeMinorTexture;
+
+uniform vec2 uFrameViewMin;
+uniform vec2 uFrameViewMax;
+uniform vec3 uPointColor;
+uniform highp int uShapeType;
+uniform int uNumCmaps;
+uniform int uCmapIndex;
+uniform float uFeatherWidth;
+uniform float uPointSize;
+uniform bool uSizeMajorMapEnabled;
+uniform bool uAreaMode;
+uniform bool uShowSelectedSource;
+uniform bool uSizeMinorMapEnabled;
+uniform bool uAreaModeMinor;
+uniform bool uCmapEnabled;
+uniform bool uOmapEnabled;
+
+uniform float uRotationAngle;
+uniform vec2 uRangeOffset;
+uniform vec2 uRangeScale;
+uniform float uScaleAdjustment;
+
+out vec3 v_colour;
+out float v_pointSize;
+out float v_orientation;
+out float v_selected;
+out float v_minorSize;
+out float v_featherWidth;
+
+
+vec4 getValueByIndexFromTexture(sampler2D texture, int index) {
+    ivec2 size = textureSize(texture, 0);
+    int row = index / size.x;
```

Comment:
will this be a potential issue of divided by zero? if so, we may need to add some error handling for dividing

Reply:
no need to worry, it will just result in an undefined pixel output.